app/channels/api_channel.rb

```diff
@@ -93,6 +137,27 @@ def zero_results data
     end
   end
 
+  def finish
+    task = Task.find(task_id)
+    unless task.scoring_open
+      dispatch_self 'errors/push', 'Finish failed: Scoring is closed'
+      return
+    end
+
+    ActiveRecord::Base.transaction do
+      task.update!(scoring_open: false)
+
+      task.contest.users.find_each do |u|
+        score = CriterionUserResult.where(user: u, criterion: task.criterions).sum(:value)
```

Comment:
@just806me
Should we raise an error if some user does not have a result for some criterion?  What do you think?